src/NoteWorkspace.ts

```diff
@@ -51,6 +51,7 @@ export class NoteWorkspace {
   static _rxTagNoAnchors = '\\#[\\w\\-\\_]+'; // used to match tags that appear within lines
   static _rxTagWithAnchors = '^\\#[\\w\\-\\_]+$'; // used to match entire words
   static _rxWikiLink = '\\[\\[[^sep\\]]+(sep[^sep\\]]+)?\\]\\]'; // [[wiki-link-regex(|with potential pipe)?]] Note: "sep" will be replaced with pipedWikiLinksSeparator on compile
+  static _rxTitle = '(?<=^( {0,3}#[^\\S\\r\\n]+))([^\\r\\n]+)';
```

Comment:
Can you add a comment that explains a bit about what this regex matches? 

I am pretty familiar with regex, but tbh, I don't quite get what all the characters before ` {0,3}#` are doing. Is this 0 to 3 space characters before a `#` ? What is the `(?<=^(` ?

And so the `#` is directly succeed by 1 or more characters that not not-space, return, or newline? I don't get why the `\\S` is in there, actually?

Reply:
I make an explanation in [here](https://regex101.com/r/g89Y3U/2) :)

Reply:
I'll submit changes after we finish our discussion and work out all the issues

I am looking forward to your reply :)